installer/scripts/service_control

```diff
@@ -16,33 +19,107 @@
 #    reload:   Reload agent configuration
 #
 
-VAR_DIR=/var/opt/microsoft/omsagent
-ETC_DIR=/etc/opt/microsoft/omsagent
-BIN_DIR=/opt/microsoft/omsagent/bin
-
-WORKSPACE_REGEX='^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$'
+OMSAGENT_START_WAIT_MAX=18
+OMSAGENT_STOP_WAIT_MAX=15
+STOP_PROCESS_WAIT_MAX=5
 
 # Space seperated list of non oms workspaces
 NON_OMS_WS="scom"
 
+# Standard Path Definitions
+
+COMMONSEGMENT=/opt/microsoft/omsagent
+BIN_DIR=$COMMONSEGMENT/bin
+ETC_DIR=/etc$COMMONSEGMENT
+VAR_DIR=/var$COMMONSEGMENT
+
+# Error codes and categories:
+
+# User configuration/parameters:
+
+INVALID_OPTION_PROVIDED=2
```

Comment:
Since your error counts will generally be smaller numbers (+1 for every workspace with an error), I would suggest making your other explicit error codes >=10.

Reply:
Yes, I'd like to use standard stuff.  Can we begin an omsagent_errors.env file so we all use the same ones?

Reply:
It may be possible to have a file like that to store constants such as error codes; in my mind, it would go under `/opt/microsoft/omsagent/bin/`.

Our files which current use constant error codes are:
- [bundle skeleton](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/installer/bundle/bundle_skel.sh); this won't be able to use a packaged file like the rest of the code might, since the bundle has to run before it can assume the file will exist. Pulling constant error codes from this file to the bundle code may require some work; perhaps it can be inserted dynamically by our [create_bundle script](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/installer/bundle/create_bundle.sh)?
- [omsadmin.sh](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/installer/scripts/omsadmin.sh); this can simply execute the error codes file to get its values
- service_control script
- [Ruby maintenance helper script](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/source/code/plugins/agent_maintenance_script.rb) - reading an error code file meant for shell will probably require a little more work here

Reply:
Okay, I have headway on this, and it passes tests.  next.